pypeit/spectrographs/keck_mosfire.py

```diff
@@ -432,3 +516,233 @@ def tweak_standard(self, wave_in, counts_in, counts_ivar_in, gpm_in, meta_table,
         else:
             return wave_in, counts_in, counts_ivar_in, gpm_in
 
+    def get_slitmask(self, filename):
+        """
+        Parse the slitmask data from a MOSFIRE file into :attr:`slitmask`, a
+        :class:`~pypeit.spectrographs.slitmask.SlitMask` object.
+
+        This can be used for multi-object slitmask, but it it's not good
+        for "LONGSLIT" nor "long2pos". Both "LONGSLIT" and "long2pos" have emtpy/incomplete
+        binTable where the slitmask data are stored.
+
+
+        Args:
+            filename (:obj:`str`):
+                Name of the file to read.
+
+        Returns:
+            :class:`~pypeit.spectrographs.slitmask.SlitMask`: The slitmask
+            data read from the file. The returned object is the same as
+            :attr:`slitmask`.
+        """
+        # Open the file
+        hdu = io.fits_open(filename)
+
+        # load slitmask info
+        targs = hdu['Target_List'].data
+        ssl = hdu['Science_Slit_List'].data
+        msl = hdu['Mechanical_Slit_List'].data
+        # some needed cleanup
+        ssl = ssl[ssl['Slit_Number'] != ' ']
+        msl = msl[msl['Slit_Number'] != ' ']
+
+        # Book keeping: Count and check that the # of objects in the SSL matches that of the MSL
+        # and if we recover the total number of CSUs
+        CSUnumslits = 46
+        numslits = np.zeros(len(ssl))
+        for i in range(len(ssl)):
+            slit = ssl[i]
+            numslits[i] = np.where(slit['Target_Name'] == msl['Target_in_Slit'])[0].size
+
+        if (numslits.sum() != CSUnumslits) and ('LONGSLIT' not in self.get_meta_value(filename, 'decker')) \
+                and ('long2pos' not in self.get_meta_value(filename, 'decker')):
+            msgs.error('The number of allocated CSU slits does not match the number of possible slits. '
+                       'Slitmask design matching not possible. Turn parameter `use_maskdesign` off')
+
+        targ_dist_center = np.array(ssl['Target_to_center_of_slit_distance'], dtype=float)
+
+        slit_centers = np.array(ssl['Slit_length'], dtype=float) / 2.
+        # Some # adjustment for long2pos
+        if 'long2pos' in self.get_meta_value(filename, 'decker'):
+            slit_centers_wrong = np.array(ssl['Slit_length'], dtype=float) / 2.
+            # correct slit length
+            ssl['Slit_length'] = '22.970', '7.010', '22.970'
```

Comment:
Are these written to the header too?  If so, maybe pull
from there in case they change some day.

Reply:
No, unfortunately these are not written in the header.  I added a method that provides these numbers, so in case we need to make some modifications in the future, it will be easier.